packages/mongo/mongo_driver.js

```diff
@@ -180,8 +180,10 @@ MongoConnection = function (url, options) {
         }
 
         // First, figure out what the current primary is, if any.
-        if (db.serverConfig._state.master)
-          self._primary = db.serverConfig._state.master.name;
+        if (db.serverConfig.isMasterDoc) {
+          self._primary = db.serverConfig.isMasterDoc.primary;
```

Comment:
Just as a note, its a bad naming convention on behalf of `mongo` actually. `isMasterDoc` is a document, not event a boolean (very misleading).

https://docs.mongodb.com/manual/reference/command/isMaster/#dbcmd.isMaster